networkx/algorithms/maximum_priority_matching.py

```diff
@@ -0,0 +1,1509 @@
+import doctest
+import logging
+import random
+import random as rd
+import time
+from typing import Dict, List, Union
+
+import networkx as nx
+
+"""
+Programmers: Roi Meshulam and Liroy Melamed
+
+This file is implemation of "Maximum Prioirity Matching" from the article that
+written by Turner, Jonathan S.
+
+more details about this algo can be found in the following link: "http://myusername.pythonanywhere.com/"
+
+"""
+
+logging.basicConfig(filename="my_logger.log", level=logging.INFO, filemode="w")
+logger = logging.getLogger()
+
+
+def find_priority_score(G: nx.Graph):
```

Comment:
This function should accept the matching as a second parameter, so that you can compare priority scores of different matchings in the same graph.

Also, in the implementation, it is not clear why you need to change the node attributes. The function should just compute the score - not change the graph.